pkg/helm/actions/auth.go

```diff
@@ -0,0 +1,125 @@
+		if err != nil {
+			return nil, fmt.Errorf("Failed to GET secret %s from %vreason %v", secretName, secretNamespace, err)
+		}
+		tlsCertBytes, found := secret.Data[tlsSecretCertKey]
+		if !found {
+			return nil, fmt.Errorf("Failed to find %s key in secret %s", tlsSecretCertKey, secretName)
+		}
+		tlsCertFile, err := writeTempFile((tlsCertBytes), tlsSecretPattern)
+		if err != nil {
+			return nil, err
+		}
+		chartPathOptions.CertFile = tlsCertFile.Name()
+		tlsFiles = append(tlsFiles, tlsCertFile)
+		tlsKeyBytes, found := secret.Data[tlsSecretKey]
+		if !found {
+			return nil, fmt.Errorf("Failed to find %s key in secret %s", tlsSecretKey, secretName)
+		}
+		tlsKeyFile, err := writeTempFile(tlsKeyBytes, tlsKeyPattern)
+		if err != nil {
+			return nil, err
+		}
+		chartPathOptions.KeyFile = tlsKeyFile.Name()
+		tlsFiles = append(tlsFiles, tlsKeyFile)
+	}
+	//set up ca certificate
+	if connectionConfig.CA != (configv1.ConfigMapNameReference{}) {
+		chartPathOptions.RepoURL = connectionConfig.URL
+		configMapName = connectionConfig.CA.Name
+		configMapNameSpace = configNamespace
+		configMap, err := coreClient.ConfigMaps(configMapNameSpace).Get(context.TODO(), configMapName, v1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Failed to GET configmap %s, reason %v", configMapName, err)
+		}
+		caCertBytes, found := configMap.Data[caBundleKey]
+		if !found {
+			return nil, fmt.Errorf("Failed to find %s key in configmap %s", caBundleKey, configMapName)
+		}
+		caCertFile, caCertGetErr := writeTempFile([]byte(caCertBytes), "cacert-*")
+		if caCertGetErr != nil {
+			return nil, caCertGetErr
+		}
+		chartPathOptions.CaFile = caCertFile.Name()
+		tlsFiles = append(tlsFiles, caCertFile)
+	}
+	return tlsFiles, nil
+}
+
+func setUpAuthenticationProject(chartPathOptions *action.ChartPathOptions, connectionConfig *v1beta1.ConnectionConfigNamespaceScoped, coreClient corev1client.CoreV1Interface, namespace string) ([]*os.File, error) {
+	tlsFiles := []*os.File{}
+	var secretNamespace, configMapName, configMapNameSpace, secretName string
+	//set up tls cert and key
+	if connectionConfig.TLSClientConfig != (configv1.SecretNameReference{}) {
+		chartPathOptions.RepoURL = connectionConfig.URL
+		secretName = connectionConfig.TLSClientConfig.Name
+		secretNamespace = namespace
+		secret, err := coreClient.Secrets(secretNamespace).Get(context.TODO(), secretName, v1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Failed to GET secret %s from %vreason %v", secretName, secretNamespace, err)
+		}
+		tlsCertBytes, found := secret.Data[tlsSecretCertKey]
+		if !found {
+			return nil, fmt.Errorf("Failed to find %s key in secret %s", tlsSecretCertKey, secretName)
+		}
+		tlsCertFile, err := writeTempFile((tlsCertBytes), tlsSecretPattern)
+		if err != nil {
+			return nil, err
+		}
+		chartPathOptions.CertFile = tlsCertFile.Name()
+		tlsFiles = append(tlsFiles, tlsCertFile)
+		tlsKeyBytes, found := secret.Data[tlsSecretKey]
```

Comment:
I would suggest to add additional validation, both for cert and private key.
Check https://github.com/openshift/console-operator/blob/master/pkg/console/subresource/route/route.go#L203-L227